CMakeLists.txt

```diff
@@ -477,14 +489,15 @@ IF (ENABLE_GCOV)
   ENDIF()
 ENDIF(ENABLE_GCOV)
 
-# Require C++11 while leaving possibility to fall back by modifying this file
+# Check for the compiler C++11 flag, if we go to requiring >= cmake 3.1.3 we could drop this
+# and just write
+# set(CMAKE_CXX_STANDARD_REQUIRED 11 CACHE INTEGER "C++ stangard must be at least 11")
```

Comment:
CXX11_NEEDED was introduced when we support compilers with and without C++11.
Now C++11 is mandatory.
For this reason, we should clean it up. I confess that I didn't place it in a good spot.
I think now it is time to solve it. We need to remove CXX11_NEEDED and then move the whole section before the MPI detection.

Reply:
I think this is outside of the scope of this PR which really is just so I and others can use llvm 5 and AST based tools while we work on qmcpack. I it in there but removed it because I expected pushback that would delay the PR.

Reply:
OK. Let me handle that with a different PR. Unfortunately, that requires another checking on multiple platforms.